addons/l10n_in_withholding/models/account_move_line.py

```diff
@@ -13,3 +13,17 @@ def _compute_withhold_tax_amount(self):
         (self - withholding_lines).l10n_in_withhold_tax_amount = False
         for line in withholding_lines:
             line.l10n_in_withhold_tax_amount = line.currency_id.round(abs(line.price_total - line.price_subtotal))
+
+    def action_l10n_in_apply_higher_tax(self):
+        for line in self:
+            updated_tax_ids = []
+            for tax in line.tax_ids:
+                if tax.l10n_in_section_id.tax_source_type == 'tcs':
+                    max_tax = max(
+                        tax.l10n_in_section_id.l10n_in_section_tax_ids,
+                        key=lambda t: t.amount
+                    )
+                    updated_tax_ids.append(max_tax.id)
+                else:
+                    updated_tax_ids.append(tax.id)
+            line.write({'tax_ids': [Command.clear()] + [Command.set(updated_tax_ids)]})
```

Comment:
Could we only do the write when something changes?

Reply:
done.